packages/generated/client-types/src/lib/models/AdminDeleteDiscountsDiscountConditionsConditionParams.ts

```diff
@@ -0,0 +1,15 @@
+/* istanbul ignore file */
+/* tslint:disable */
+/* eslint-disable */
+import { SetRelation, Merge } from "../core/ModelUtils"
+
+export interface AdminDeleteDiscountsDiscountConditionsConditionParams {
```

Comment:
question: All those interface shouldn't extends `FindParams` instead? Maybe not since it is regenerated if we change something. wdyt?

Reply:
Sadly, FindParams does not exist in our OAS realm. Parameter definition in OAS does not have the same composition capabilities as TS. That is why we have a lot of repetition of `fields` and `expand` in our OAS. As an [example](https://github.com/medusajs/medusa/blob/feat/codegen-build-commit/packages/medusa/src/api/routes/store/orders/get-order.ts#L10-L13).

In order to achieve bundling of query Params into a type, we had to customize the codegen lib by introducing our own `x-codegen.queryParams` OAS extension. You can find the code [here](https://github.com/medusajs/medusa/blob/feat/codegen-build-commit/packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModels.ts#L35-L117).